lib/OBSWebSocket.js

```diff
@@ -37,20 +35,20 @@ class OBSWebSocket extends Socket {
       const messageId = args['message-id'] = generateMessageId(); // eslint-disable-line no-multi-assign
 
       // Submit the request to the websocket.
-      log.debug('[Core:send]', messageId, requestType, args);
+      debug('[send] %s %s %O', messageId, requestType, args);
       this._socket.send(JSON.stringify(args));
```

Comment:
Ah okay, it's because of the automatic call to `this.connect` that happens in the constructor for the `Socket` class. I don't think there's a good way for us to return this promise.

It might be best to get rid of this automatic `connect` and instead mandate that the consumer calls `.connect` themselves. What are your thoughts on that?